nx/lib/nx.ex

```diff
@@ -6658,6 +6658,218 @@ defmodule Nx do
     impl!(tensor).cholesky(out, tensor)
   end
 
+  @doc """
+  Calculates the p-norm of a tensor.
+
+  For the 0-norm, the norm is the number of non-zero elements in the tensor.
+
+  ## Options
+
+    * `:axes` - defines the axes upon which the norm will be calculated. Default: `nil`.
+    * `:keep_axes` - defines if the resulting tensor should have the same dimensions as the input. Default: `false`.
+    * `:ord` - defines which norm will be calculated according to the table below. Default: `2`
+
+  | ord          | 2-D                                       | 1-D                               |
+  | ------------ | ----------------------------------------- | --------------------------------- |
+  | `nil`        | Frobenius norm                            | 2-norm                            |
+  | `:frobenius` | Frobenius norm                            | -                                 |
+  | `:nuclear`   | Nuclear norm (not implemented)            | -                                 |
+  | `:inf`       | `max(sum(abs(x), axes: 1))`               | `max(abs(x))`                     |
+  | `:neg_inf`   | `min(sum(abs(x), axes: 1))`               | `min(abs(x))`                     |
+  | 0            | -                                         | Number of non-zero elements       |
+  | 1            | `max(sum(abs(x), axes: 0))`               | as below                          |
+  | -1           | `min(sum(abs(x), axes: 0))`               | as below                          |
+  | 2            | 2-norm                                    | as below                          |
+  | -2           | smallest singular value (not implemented) | as below                          |
+  | other        | -                                         | power(sum(power(abs(x), p)), 1/p) |
+
+  ## Examples
+
+    ### Vector norms
+
+    iex> Nx.norm(Nx.tensor([3, 4]))
```

Comment:
Make sure the examples are indented with 4 spaces from the heredoc beginning indentation, as that's required by Markdown. Otherwise they won't be highlighted as code snippets. TL;DR - indent all examples by two spaces. :)